### ISSUE TITLE
Issue: METS is invalid due to premis:creatingApplication misplaced

### ISSUE THREAD

**Issue by sromkey** [artefactual/archivematica#907]
In qa.1.7 we have added the PREMIS element creatingApplication to record the date an object was created. It needs to be placed in the METS file before the premis:objectCharacteristicsExtension, rather than after, in order to pass PREMIS in METS validation.

I have examples of how it is now as well as how it would look fixed.

**Comment by jrwdunham** [comment]
Hey @sromkey https://github.com/artefactual/archivematica/pull/913 should fix this.

I created an AIP using it and used metsrw's `validate` function on it and confirmed that it passes the METS/PREMIS XSD-based validation.

However, I did also notice another potential issue when running that validation: some PREMIS:EVENTs have 3 PREMIS:AGENTs while others (e.g., the events for the transfer's METS.xml file) only have 2: they are missing the `<premis:linkingAgentIdentifierType>Archivematica user pk` agent. This is invalid given our Schematron file at https://github.com/artefactual-labs/mets-reader-writer/blob/master/metsrw/resources/archivematica_mets_schematron.xml#L125. Is this a problem with this Schematron file or is it an actual problem with the METS file we are constructing?

The error msg:

    Schematron Error(s):
    1. Each amdSec MUST contain 3 digiprovMD elements that contain
                mdWraps of MDTYPE PREMIS:AGENT
       test: count(m:digiprovMD/m:mdWrap[@MDTYPE = 'PREMIS:AGENT']) = 3
       location: /*[local-name()='mets' and namespace-uri()='http://www.loc.gov/METS/']/*[local-name()='amdSec' and namespace-uri()='http://www.loc.gov/METS/'][2]
